addon/components/kinetic-form.js

```diff
@@ -106,34 +105,35 @@ export default Component.extend({
 
   validateForm() {
     let changeset = get(this, 'changeset');
-    return changeset.validate()
-      .then(() => set(this, 'showErrors', false))
-      .then(() => changeset)
-      .catch(() => set(this, 'showErrors', true));
-  },
-
-  submitForm() {
-    let changeset = get(this, 'changeset');
-    this.validateForm(changeset).then(() => {
+    return changeset.validate().then(() => {
       if (get(changeset, 'isInvalid')) {
         set(this, 'showErrors', true);
         return;
       }
       set(this, 'showErrors', false);
-      get(this, 'onSubmit')(changeset);
     });
   },
 
+  validateAndNotifySubmit() {
+    return this.validateForm()
+      .then(() => get(this, 'onSubmit')(this.get('changeset')));
+  },
+
+  validateAndNotifyUpdate() {
```

Comment:
@sukima this won't ever fire successfully unless the entire form is complete.  I think when we update, we are unable to validate at this time.